### PR TITLE
Tests/optimiser

### PR DESCRIPTION
**Pull request by Tanguylo**
* **What kind of changes does this PR introduce?** 
- [x] performance 

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
- [x] No

* **Other information**:
Implement tests for optimizer.
Some elements in code optimizer are not tested here because they do not seem to be used. Should they be removed ?

### CODE REVIEW

File: dessia_common/optimization.py [hunk 1]
```diff
@@ -124,33 +121,41 @@ def cma_bounds(self):
...
 
         model = self.instantiate_model(attributes_values)
-
         return model, fx_opt
+
+    def optimize_cma_then_gradient(self, method: str = 'L-BFGS-B'):
+        model_cma, fx_cma = self.optimize_cma()
+        x0 = npy.array([getattr(model_cma, attr.attribute_name) for attr in self.optimization_bounds])
+        model, fx_opt = self.optimize_gradient(method=method, x0=x0)
+        best_fx = npy.min([fx_opt, fx_cma])
```
Comment (masfaraud) [inline dessia_common/optimization.py:157]: why not just if fx_opt < fx_cma:

File: dessia_common/optimization.py [hunk 2]
```diff
@@ -124,33 +121,41 @@ def cma_bounds(self):
...
+        result = scipy.optimize.minimize(self.objective_from_dimensionless_vector, x0, bounds=bounds, method=method)
 
+        attributes_values = self.vector_to_attributes_values(self.dimensionless_vector_to_vector(result.x))
         model = self.instantiate_model(attributes_values)
         return model, result.fun
 
-    def optimize_cma(self):
+    def _init_gradient(self, x0: List[float] = None):
+        if x0 is None:
+            return npy.random.random(self.number_parameters)
```
Comment (masfaraud) [inline dessia_common/optimization.py:136]: a little bit overkill. why not a generic method random_x ?

**Review by masfaraud — CHANGES_REQUESTED** [review]

File: dessia_common/optimization.py (hunk 1 above)
Comment (Tanguylo) [reply dessia_common/optimization.py:157]: Modification made:

> model, best_fx = self.optimize_gradient(method=method, x0=x0)
>  if fx_cma <= best_fx:
>       model = model_cma
>       best_fx = fx_cma
> return model, best_fx

File: dessia_common/optimization.py (hunk 2 above)
Comment (Tanguylo) [reply dessia_common/optimization.py]: Merged method _init_gradient into optimize_gradient

**Review by masfaraud — APPROVED** [review]

**Pull request merged**